.github/scripts/transform_dbt_test_results.py

```diff
@@ -1129,36 +1171,57 @@ def get_test_categories(
                                 and par.deactivat is null
                         """
 
+            test_results_filter = ""
+            if townships:
+                test_results_filter = (
+                    " where leg.user1 in "
+                    "(" + ",".join(f"'{code}'" for code in townships) + ")"
+                )
+
             test_results_query = (
                 f"{test_results_select} "
                 f"from {test_results_relation_name} as test_results "
                 f"{test_results_join}"
+                f"{test_results_filter}"
             )
             cursor.execute(test_results_query)
             query_results = cursor.fetchall()
             if len(query_results) == 0:
-                raise ValueError(
-                    f"Test {test_name} has status '{status!r}' but no failing "
+                msg = (
+                    f"Test {test_name} has status {status!r} but no failing "
                     "rows in Athena"
                 )
-
-            # Add custom fields to query results that we don't expect to be
-            # included in the response
-            failing_rows = [
-                {
-                    TEST_NAME_FIELD: test_name,
-                    DESCRIPTION_FIELD: test_description,
-                    DOCS_URL_FIELD: test_docs_url,
-                    SOURCE_TABLE_FIELD: tablename,
-                    **row,
-                }
-                for row in query_results
-            ]
-            test_result = TestResult(
-                status=Status(status),
-                failing_rows=failing_rows,
-                **base_result_kwargs,
-            )
+                if townships:
+                    # Missing rows are most likely due to the township filter,
+                    # so print a warning and skip
+                    print(msg + ", possibly due to the township filter")
+                    test_result = TestResult(
+                        status=Status.PASS,
+                        failing_rows=[],
+                        **base_result_kwargs,
+                    )
```

Comment:
This block only gets executed if a test failed, but none of its failing rows are in the township(s) specified by the `--township` option. In this case, we override the results to mimic the case in which the tests passed.

Note that there is technically a chance that this block could execute if a `--township` filter is present but there is some other unexpected problem with the results metadata. In this case, I expect that some piece of logic further down the call stack will raise a different error, and my hope is that the "possibly" modifier in the warning log line will be enough to flag for an end user that a problem occurred around the time of this block's execution,